quixstreams/platforms/quix/api.py

```diff
@@ -63,24 +61,47 @@ def request(self, method, url, **kwargs):
     @property
     def default_workspace_id(self) -> str:
         if not self._default_workspace_id:
-            raise self.UndefinedQuixWorkspaceId("You must provide a Quix Workspace ID")
+            raise UndefinedQuixWorkspaceId("You must provide a Quix Workspace ID")
         return self._default_workspace_id
 
     @default_workspace_id.setter
     def default_workspace_id(self, value):
         self._default_workspace_id = value
 
+    def _response_handler(self, r: requests.Response, *args, **kwargs):
+        """
+        Custom callback/hook that is called after receiving a request.Response
+
+        Catches non-200's and passes both the original exception and the Response body.
+
+        Note: *args and **kwargs expected for hook
+        """
+        try:
+            r.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            try:
+                reason_text = e.response.json()
+            except requests.exceptions.JSONDecodeError:
+                reason_text = e.response.text
+
+            raise QuixApiRequestFailure(
+                {
+                    "response_error": e,  # response error
+                    "response_body": reason_text,  # error message in the API response
+                }
```

Comment:
1. Let's catch only `requests.exceptions.HTTPError` here, because this is what is raised from `raise_for_status()`
2. Instead of passing a dictionary here, let's pass an error message as a string to make it more pythonic and limit the output. For example: `{status code} Error for url "{url}": {response_body}`:  
`400 Error for URL "https://example.com": {"error": "Error message from the server"}`